src/benchInsert.c

```diff
@@ -1710,6 +1711,151 @@ void loadChildTableInfo(threadInfo* pThreadInfo) {
     tmfree(buf);
 }
 
+// create conn again
+int32_t reCreateConn(threadInfo * pThreadInfo) {
+    // single
+    bool single = true;
+    if (pThreadInfo->dbInfo->superTbls->size > 1) {
+        single = false;
+    }
+
+    //
+    // retry stmt2 init 
+    //
+
+    // stmt2 close
+    if (pThreadInfo->conn->stmt2) {
+        taos_stmt2_close(pThreadInfo->conn->stmt2);
+        pThreadInfo->conn->stmt2 = NULL;
+    }
+
+    // retry stmt3 init , maybe success
+    pThreadInfo->conn->stmt2 = initStmt2(pThreadInfo->conn->taos, single);
+    if (pThreadInfo->conn->stmt2) {
+        succPrint("%s", "reCreateConn first taos_stmt2_init() success and return.\n");
+        // select db 
+        if (taos_select_db(pThreadInfo->conn->taos, pThreadInfo->dbInfo->dbName)) {
+            errorPrint("taos select database(%s) failed\n", pThreadInfo->dbInfo->dbName);
+            return -1;
+        }
+        return 0;
+    }
+
+    //
+    // close old
+    //
+    closeBenchConn(pThreadInfo->conn);
+    pThreadInfo->conn = NULL;
+
+    //
+    // create new
+    //
+
+    // conn
+    pThreadInfo->conn = initBenchConn();
+    if (pThreadInfo->conn == NULL) {
+        errorPrint("%s", "reCreateConn initBenchConn failed.");
+        return -1;
+    }
+    // stmt2
+    pThreadInfo->conn->stmt2 = initStmt2(pThreadInfo->conn->taos, single);
+    if (NULL == pThreadInfo->conn->stmt2) {
+        errorPrint("reCreateConn taos_stmt2_init() failed, reason: %s\n", taos_errstr(NULL));
+        return -1;
+    } else {
+        succPrint("%s", "reCreateConn second taos_stmt2_init() success.\n");
+    }
+
+    // select db 

```

Comment:
select db after conn init

Reply:
already select db after conn init ,  this may be misunderstood, I change anther method